src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -118,7 +118,7 @@ def repo_status_icon(status, details = nil)
 
     description = REPO_STATUS_DESCRIPTIONS[status] || 'Unknown state of repository'
     description = 'State needs recalculations, former state was: ' + description if outdated
-    description << " (" + details + ")" if details
+    description += " (" + details + ")" if details
```

Comment:
They aren't the same thing, you can see it in this article: 
http://rubyquicktips.com/post/1197849149/string-concatenation-vs

What is happening is that we are modifing indirectly `REPO_STATUS_DESCRIPTIONS` constant. The combination of to asign `REPO_STATUS_DESCRIPTIONS[status]` to a variable ends up being a reference to the constant value and to use `<<` that changes both objects (variable and the value in the hash constant) and the fact that in production the helpers are not reloaded, leads us that the constant value is shared between requests.

Those lines are the culprit

``` ruby
# lines 119-121 of webui/webui_helper.rb

description = REPO_STATUS_DESCRIPTIONS[status] || 'Unknown state of repository'
description = 'State needs recalculations, former state was: ' + description if outdated
description << " (" + details + ")" if details 
```